packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt

```diff
@@ -117,21 +125,24 @@ public class AuthCognito : FlutterPlugin, MethodCallHandler {
     try {
       data = checkData(checkArguments(call.arguments));
     } catch(e: Exception) {
-      prepareError(result, e, FlutterAuthFailureMessage.CASTING.toString())
+      return prepareError(result, "AmplifyCastingException", createSerializedError(
```

Comment:
Similarly here `AmplifyCastException` is not useful for developers to catch specifically.It should still be an AuthException here and the underlyingException will tell what really happened.